src/v/kafka/server/protocol_utils.cc

```diff
@@ -64,9 +68,48 @@ parse_header(ss::input_stream<char>& src) {
     header.client_id = std::string_view(
       header.client_id_buffer.get(), header.client_id_buffer.size());
     validate_utf8(*header.client_id);
+
+    /// Conditionally handle v1 (flex) header
+    if (!flex_versions::is_api_in_schema(header.key)) {
+        /// Either user provided unsupported API key or schema is too old to
+        /// know about particular version being flex or not. In either case
+        /// interpret request as not flex and continue
```

Comment:
Think it's worth adding a debug log here?

Reply:
Eventually the code will hit this and throw: https://github.com/redpanda-data/redpanda/blob/dev/src/v/kafka/server/requests.cc#L337

I think its worth to clarify that in the comment though